_posts/2019-12-19-codeception-4.markdown

```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title: "Codeception 4.0 Released"
+date: 2019-12-18 01:03:50
+---
+
+We are finally here!
+
+Codeception 4.0 is released and this means a lot for us!
+
+After long months on spitting core into modules, publishing new packages and setting up build pipelines, we are proud to show you the result.
+
+As you know - Codeception 4 changes the way it is installed. From now on, Codeception package won't include any modules by default and you will need to install them manually. This can be a bit annoying at first but in the end, this will allow you to upgrade each package individually and not relying on the core.
+
+In Codeception 4 we also **added support for Symfony 5**, so if you plan to use the latest Symfony components - you need to upgrade today.
+
+## Upgrade to Codeception 4
+
+To start upgrading to Codeception 4 bump a version in `composer.json` and run `composer update`.
+
+```
+"codeception/codeception": "^4.0"
+```
+
+To simplify the upgrading process **we introduced an upgrade script** which will go through your config file and scan for all included modules. Corresponding packages will be added to composer.json, so you could install all Codeception modules you need. 
+
+Run upgrade script on current project:
+
+```
+php vendor/bin/codecept init upgrade4
+```
+
+The script will upgrade your `composer.json` file and it is recommended to run `composer update` immediately, so new modules will be installed. 
+
+If you use several Codeception configs for multiple bundles you will need to execute upgrade script for each of them:
+
+```
+php vendor/bin/codecept init upgrade4 -c src/app/frontend/
+```
+
+Congratulations! An upgrade is done. Now you will be able to use each module individually.
+
+**Thanks to Gintautas Miselis @Naktibalda for splitting the core and working Symfony support**
```

Comment:
`working Symfony support` is not gramatically correct either.

Reply:
Ok!